packages/design-system-react/src/components/button-base/ButtonBase.types.ts

```diff
@@ -0,0 +1,70 @@
+import type { ComponentProps } from 'react';
+
+import type { IconName, IconProps } from '../icon';
+
+export enum ButtonBaseSize {
+  Sm = 'sm',
+  Md = 'md',
+  Lg = 'lg',
+}
+
+export type ButtonBaseProps = ComponentProps<'button'> & {
```

Comment:
Make sure all types that are optional use the "optional prop" language present in other components